code/modules/projectiles/projectile/energy.dm

```diff
@@ -159,44 +159,34 @@
 /obj/item/projectile/energy/bolt/large
 	damage = 20
 
-/obj/item/projectile/energy/tesla_revolver
+/obj/item/projectile/energy/tesla
 	name = "tesla bolt"
 	icon_state = "tesla_projectile"
 	impact_effect_type = /obj/effect/overlay/temp/impact_effect/blue_laser
 	var/chain
 
-/obj/item/projectile/energy/tesla_revolver/fire(setAngle)
+/obj/item/projectile/energy/tesla/fire(setAngle)
 	if(firer)
 		chain = firer.Beam(src, icon_state = "lightning[rand(1, 12)]", time = INFINITY, maxdistance = INFINITY)
 	..()
 
-/obj/item/projectile/energy/tesla_revolver/on_hit(atom/target)
-	. = ..()
-	if(isliving(target))
-		tesla_zap(src, 3, 10000)
-	qdel(src)
-
-/obj/item/projectile/energy/tesla_revolver/Destroy()
+/obj/item/projectile/energy/tesla/Destroy()
 	qdel(chain)
 	return ..()
 
+/obj/item/projectile/energy/tesla/revolver
+	name = "energy orb"
 
-/obj/item/projectile/energy/tesla_cannon
-	name = "tesla bolt"
-	icon_state = "tesla_projectile"
-	impact_effect_type = /obj/effect/overlay/temp/impact_effect/blue_laser
-	var/chain
+/obj/item/projectile/energy/tesla/revolver/on_hit(atom/target)
+	. = ..()
+	if(isliving(target))
+		tesla_zap(target, 3, 10000)
+	qdel(src)
 
-/obj/item/projectile/energy/tesla_cannon/fire(setAngle)
-	if(firer)
-		chain = firer.Beam(src, icon_state = "lightning[rand(1, 12)]", time = INFINITY, maxdistance = INFINITY)
-	..()
+/obj/item/projectile/energy/tesla/cannon
+	name = "tesla orb"
 
-/obj/item/projectile/energy/tesla_cannon/on_hit(atom/target)
+/obj/item/projectile/energy/tesla/cannon/on_hit(atom/target)
 	. = ..()
-	tesla_zap(src, 3, 10000, explosive = FALSE, stun_mobs = FALSE)
+	tesla_zap(target, 3, 10000, explosive = FALSE, stun_mobs = FALSE)

```

Comment:
@Iamgoofball how so? It's zapping from whatever it hit, instead of itself. That kinda makes sense since the energy is going into whatever got shot, instead of "This is a tesla orb that explodes with lightning whenever it hits something"
also considering the projectile is deleted that just makes the beam weird I believe.